### PR TITLE
feat: Add custom components

### PR DESCRIPTION
**Pull request by AkshatGKMit**
- Added icon, icon button, status bar and gradient screen component
- Fixed android build errors by degrading sdk versions
- Utilized themes in a more optimized manner in styles

### CODE REVIEW

File: App.tsx [hunk 1]
```diff
@@ -1,10 +1,34 @@
...
+
+const Main = () => {
+  const { theme } = useContext(SettingsContext);
+
+  const globalStyles = GlobalThemedStyles(theme);
+
+  return (
+    <>
+      <PlatformDependentStatusBar />
+      <SafeAreaView style={[globalStyles.screen]}>
```
Comment (adiTrivsme) [inline App.tsx:28]: why are there two instances of "SafeArea"?

File: android/app/src/main/java/com/tuneify/MainApplication.kt [hunk 2]
```diff
@@ -9,8 +9,8 @@ import com.facebook.react.ReactPackage
 import com.facebook.react.defaults.DefaultNewArchitectureEntryPoint.load
 import com.facebook.react.defaults.DefaultReactHost.getDefaultReactHost
 import com.facebook.react.defaults.DefaultReactNativeHost
-import com.facebook.react.soloader.OpenSourceMergedSoMapping
 import com.facebook.soloader.SoLoader
+import com.tuneify.BuildConfig
```
Comment (adiTrivsme) [inline android/app/src/main/java/com/tuneify/MainApplication.kt:13]: explain the reason for this?

File: src/components/icon/Icon.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+};
+
+const Icon = ({ family, name, style }: IconProps) => {
+  const { theme } = useContext(SettingsContext);
+
+  const styles = ThemedStyles(theme);
+
+  const SelectedIcon = iconFamilies[family];
+
+  return SelectedIcon ? (
```
Comment (adiTrivsme) [inline src/components/icon/Icon.tsx]: use `if {...}` statement

File: src/components/icon/style.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { useMemo } from 'react';
+import { StyleSheet } from 'react-native';
+
+const ThemedStyles = (theme: ThemeColors) => {
+  return useMemo(() => {
+    return StyleSheet.create({
+      icon: {
+        color: theme.defaultIconColor,
+        fontSize: 18,
```
Comment (adiTrivsme) [inline src/components/icon/style.ts]: use fonts from constants

File: src/components/iconButton/IconButton.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
+import { useContext, useMemo } from 'react';
+import { TouchableHighlight } from 'react-native';
+
+import SettingsContext from '@config/SettingsContext';
+
+import Icon from '@components/icon';
```
Comment (adiTrivsme) [inline src/components/iconButton/IconButton.tsx:4]: fix order 

import Icon from '@components/icon';

import SettingsContext from '@config/SettingsContext';

......

File: src/config/platformDependentStatusBar/PlatformDependentStatusBar.tsx [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+import { isIos } from '@constants';
+import ThemedStyles from './styles';
+
+const PlatformDependentStatusBar = () => {
+  const { theme } = useContext(SettingsContext);
+
+  const styles = ThemedStyles(theme);
+
+  if (!isIos) {
```
Comment (adiTrivsme) [inline src/config/platformDependentStatusBar/PlatformDependentStatusBar.tsx:14]: mention the reason of not using StatusBar in iOS

File: src/constants/index.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+  Entypo: 'Entypo',
+  Feather: 'Feather',
+  FontAwesome: 'FontAwesome',
+  Fontisto: 'Fontisto',
+  Ionicons: 'Ionicons',
+  MaterialCommunityIcons: 'MaterialCommunityIcons',
+  MaterialIcons: 'MaterialIcons',
+  Octicons: 'Octicons',
+  SimpleLineIcons: 'SimpleLineIcons',
+} as const;
```
Comment (adiTrivsme) [inline src/constants/index.ts:16]: mention the reason of using `as const`

File: android/app/src/main/java/com/tuneify/MainApplication.kt (hunk 2 above)
Comment (AkshatGKMit) [reply android/app/src/main/java/com/tuneify/MainApplication.kt:13]: while building its android app
i was encountering error of cmake build failed and and debug kotlin failed somewhat similar to this
`[Unresolved reference: R and BuildConfig in React Native Android Project with Kotlin]`

so i opened the android folder in android studio,
i saw build config was user 4 times (line: 27, 29, 30, 39) in this file but it was not available, It was suggesting to import it

so this is the reason

File: App.tsx (hunk 1 above)
Comment (AkshatGKMit) [reply App.tsx:28]: The main reason of using 2 safe areas is due to ios status bar issue
Status bar component did not work properly in ios, its background color did not change

so for ios i assigned a view that take status bar height(from safe area provider wrapped in App)
and then provide color to view
also thats why this component is outside the safe are view

File: src/config/platformDependentStatusBar/PlatformDependentStatusBar.tsx (hunk 6 above)
Comment (AkshatGKMit) [reply src/config/platformDependentStatusBar/PlatformDependentStatusBar.tsx:14]: same as above

File: src/constants/index.ts (hunk 7 above)
Comment (AkshatGKMit) [reply src/constants/index.ts:16]: so that it acts as readonly

File: App.tsx (hunk 1 above)
Comment (adiTrivsme) [reply App.tsx:28]: If it only solving status bar issue, a more meaningful name to the component can be used

File: android/app/src/main/java/com/tuneify/MainApplication.kt (hunk 2 above)
Comment (adiTrivsme) [reply android/app/src/main/java/com/tuneify/MainApplication.kt:13]: is it because of the RN version??

File: src/components/icon/Icon.tsx (hunk 3 above)
Comment (adiTrivsme) [reply src/components/icon/Icon.tsx]: RESOLVE CONVERSATION IF THE ISSUE IS ADDRESSED

**Review by adiTrivsme — APPROVED** [review]

**Pull request merged**